user-app/scripts/sentiment.py

```diff
@@ -0,0 +1,223 @@
+        self.labels = ['Negative', 'Neutral', 'Positive']
+
+
+    def cleanData(self):
+        '''Some genereal data preperation, remove all symbols and numbers.
+        transforms all text to lower case and removes redundant and trailing 
+        spaces.
+        '''
+        df = self.dtaframe.copy()
+        colname = self.colname
+
+        import pandas as pd
+        pd.options.mode.chained_assignment = None  # default='warn'
+
+        # supress future warnings
+        import warnings
+        warnings.simplefilter(action='ignore', category=FutureWarning)
+
+
+        # set values to srting
+        df[colname] = df[colname].astype(str)
+
+        # remove all line breaks
+        df[colname] = df[colname].str.replace(r'\n',' ')
+
+        # all to lowercase 
+        df[colname] = df[colname].str.lower()
+
+        # remove all symbols and numbers
+        df[colname] = df[colname].str.replace('[^A-záéíóúýçäëïöüÿàèìòùâêîôûãõñ]', ' ')
+
+        while df[colname].str.find('  ').sum() > 0:
+            df[colname] = df[colname].str.replace('  ', ' ')
+
+        return df
+
+     
+     
+    def sentAnalysis(self,model_name,likert): 
+        '''This defenitions scores sentiment based on a likert scale. This is either on a 3 point or 
+        5 point scale (default is 3 point), for the 3 point scale a twitter based model is used:
+        https://huggingface.co/btjiong/robbert-twitter-sentiment?text=a. This model is spicificly 
+        trained for dutch text. 
+        The 5 point scale uses a model that is multilingual:
+        https://huggingface.co/nlptown/bert-base-multilingual-uncased-sentiment?text=a.
+        '''   
+        from transformers import AutoTokenizer, AutoModelForSequenceClassification
+        from scipy.special import softmax
+        df = self.dtaframe.copy()
+        colname = self.colname
+        labels = self.labels
+
+        # set model
+        model = AutoModelForSequenceClassification.from_pretrained(model_name)
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+  
+        # labels toevoegen als kolommen
+        for l in labels:
+            df[l] = 0.00
+
+        for i in df.index:
+            # skip all text that have less then 30 or more than 900 characters
+            if len(df[colname][i]) > 15 and len(df[colname][i]) < 900:
+                
+                sent_text = df[colname][i]
+
+                # tokinize
+                encoded_text = tokenizer(sent_text , return_tensors='pt')
+
+                # sentiment analysis
+                output = model(**encoded_text)
+
+                # get scores from sentiment
+                scores = output[0][0].detach().numpy()
+                scores = softmax(scores)
+                
+                # if model uses a 5 points scale, recode to 3 points
+                if likert == 5:
+                    import numpy as np
+                    scores = np.array([scores[0:2].sum()]+[scores[2]]+[scores[3:5].sum()])
```

Comment:
Not sure if I understand correctly, but when likert == 5 you will have 5 scores right? so scores will contain values [0:4]